test/images/agnhost/grpc-health-checking/grpc-health-checking.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package grpchealthchecking offers a tiny grpc health checking endpoint.
+package grpchealthchecking
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+)
+
+// CmdGrpcHealthChecking is used by agnhost Cobra.
+var CmdGrpcHealthChecking = &cobra.Command{
+	Use:   "grpc-health-checking",
+	Short: "Starts a simple grpc health checking endpoint",
+	Long:  "Starts a simple grpc health checking endpoint --port to serve on. The endpoint returns SERVING for the first --delay-unhealthy-sec, and NOT_SERVING after this.",
+	Args:  cobra.MaximumNArgs(0),
+	Run:   main,
+}
+
+var (
+	port              int
+	delayUnhealthySec int
+)
+
+func init() {
+	CmdGrpcHealthChecking.Flags().IntVar(&port, "port", 5000, "Port number.")
+	CmdGrpcHealthChecking.Flags().IntVar(&delayUnhealthySec, "delay-unhealthy-sec", 10, "Number of seconds to delay before start reporting unhealthy.")
```

Comment:
time based test use to be flaky, maybe adding some handler that turns it healthy or unhealthy is a better solution to have more control

Reply:
I was following `liveness` example, I agree this is potentially flaky. The nature of probes tests, however, makes it flaky to write non-flaky tests. It's hard to know deterministically when probes start either. So with enough time buffers, this should be ok. 

lmk if you have specific suggestions how to make it better, I'm open for ideas

Reply:
I'm not very familiar with probe tests but I was thinking in something ala

```
start pod probe:5000 healthy
curl pod:8080/set_probe?unhealthy
```

Reply:
make sense. Implemented